### ISSUE TITLE
[dpc++] exclusive_scan (in-place) incorrect results 

### ISSUE THREAD

**Issue by abagusetty** [oneapi-src/oneDPL#514]
Incorrect results when using `exclusive_scan` with device_policy. Specifically with in-place scan. 

```
#define _GLIBCXX_USE_TBB_PAR_BACKEND 0 // libstdc++10

#include <cassert>

#include <CL/sycl.hpp>
#include <oneapi/dpl/execution>
#include <oneapi/dpl/algorithm>

#include <algorithm>
#include <vector>
#include <iostream>

#include <numeric> // std::inclusive_scan, exclusive_scan
#include <functional>

int main()
{
  std::vector<int> v{3, 1, 4, 1, 5, 9, 2, 6};

  sycl::queue syclQue(sycl::gpu_selector{});
  int* excl_input_dev = sycl::malloc_device<int>(10, syclQue);
  syclQue.memcpy(excl_input_dev, v.data(), v.size()*sizeof(int)).wait();  

  // Exclusive scan (in-place, incorrect results)
  std::exclusive_scan( v.begin(),
                       v.end(),
                       v.begin(),
                       0 );
  oneapi::dpl::exclusive_scan( oneapi::dpl::execution::make_device_policy(syclQue),
                               excl_input_dev,
                               excl_input_dev + v.size(),
                               excl_input_dev,
                               0 );
  int* excl_result_host = new int[v.size()];
  syclQue.memcpy(excl_result_host, excl_input_dev, v.size()*sizeof(int)).wait();

  for (int i=0; i<v.size(); i++) {
    assert( v[i] == excl_result_host[i] );
  }
  delete[] excl_result_host;
  sycl::free(excl_input_dev, syclQue);

  return 0;
}
```

**Comment by SergeyKopienko** [comment]
Thank you for reporting, we will take a look.

**Comment by zjin-lcf** [comment]
Can you please fix the bug ?  It took developers and researchers quite a while to pinpoint the bug in a program that calls std::exclusive_scan.  

Thanks.

**Comment by zjin-lcf** [comment]
@ValentinaKats 
Can you please label it with a 'bug' tag ?  Thanks.

**Comment by abagusetty** [comment]
Thanks to @MikeDvorskiy for the resolving the issue. @zjin-lcf Fix was implemented in 8f9190b, 7595b51. Closing this issue.
